src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -0,0 +1,332 @@
+     *
+     * @param input Raw user input string
+     * @return String array [command, parameters]
+     */
+    private String[] splitInputIntoCommandAndParams(String input) {
+        String[] commandAndParams = new String[2];
+        final String[] inputSplit = input.trim().split(" ", 2);
+        //command string
+        commandAndParams[0] = inputSplit[0];
+        //param string; if not given, set to EMPTY for error handling
+        commandAndParams[1] = (inputSplit.length >= 2) ? inputSplit[1] : EMPTY;
+        return commandAndParams;
+    }
+
+    private boolean hasRequiredParams(String params, String... prefixes) {
+        for (String prefix : prefixes) {
+            if (!params.toLowerCase().contains(prefix + Command.COMMAND_PREFIX_DELIMITER)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private String extractItemTypePrefix(String params) throws ItemNotSpecifiedException {
+        boolean isExercise =
+                params.toLowerCase().contains(Command.COMMAND_PREFIX_EXERCISE
+                        + Command.COMMAND_PREFIX_DELIMITER);
+        boolean isFood =
+                params.toLowerCase().contains(Command.COMMAND_PREFIX_FOOD
+                        + Command.COMMAND_PREFIX_DELIMITER);
+
+        if (isExercise && isFood || !isExercise && !isFood) {
+            throw new ItemNotSpecifiedException(); //cannot be both and cannot be neither
+        } else if (isExercise) {
+            return Command.COMMAND_PREFIX_EXERCISE;
+        } else {
+            return Command.COMMAND_PREFIX_FOOD;
+        }
+    }
+
+    private String extractRelevantParameter(String params) {
+        if (params.contains(Command.COMMAND_PREFIX_DELIMITER)) {
+            return params.substring(0, params.indexOf(Command.COMMAND_PREFIX_DELIMITER) - 1).trim();
+        } else {
+            return params;
+        }
+    }
+
+    private String extractItemDescription(String params, String prefix) throws ParamInvalidException {
+        try {
+            String stringAfterPrefix = params.trim().split(prefix + Command.COMMAND_PREFIX_DELIMITER, 2)[1];
+            String description = extractRelevantParameter(stringAfterPrefix);
+            if (description.equals(EMPTY)) {
+                throw new ParamInvalidException(MESSAGE_ERROR_NO_DESCRIPTION);
+            }
+            return description;
+        } catch (IndexOutOfBoundsException e) {
+            throw new ParamInvalidException(MESSAGE_ERROR_NO_DESCRIPTION);
+        }
+    }
+
+    private int extractItemCalories(String params) throws ParamInvalidException {
+        try {
+            if (params.contains(Command.COMMAND_PREFIX_CALORIES + Command.COMMAND_PREFIX_DELIMITER)) {
+                String stringAfterPrefix =
+                        params.substring(params.indexOf(Command.COMMAND_PREFIX_CALORIES
+                                + Command.COMMAND_PREFIX_DELIMITER)
+                                + 2);
```

Comment:
Hi, may I know why there is a `+ 2` here :)
If this is a magic literal then probably need to extract it out too

Reply:
It's so that the substring starts from the first letter after the delimiter! Like in `h/23`, the substring returned will be `23`. But yea, I was very bothered by this `+ 2` too,  I think I will just use `split` instead! Thanks!